lisp/transient.el

```diff
@@ -92,8 +92,14 @@
 (defmacro transient--with-emergency-exit (&rest body)
   (declare (indent defun))
   `(condition-case err
-       ,(macroexp-progn body)
-     (error (transient--emergency-exit err))))
+       (let ((debugger #'transient--exit-and-debug))
+         ,(macroexp-progn body))
+     ((debug error)
+      (transient--emergency-exit err))))
+
+(defun transient--exit-and-debug (&rest args)
+  (transient--emergency-exit)
+  (debug args))
```

Comment:
Do you need to call `transient--emergency-exit` before `debug` when you're also adding it to `debugger-mode-hook`? Actually, since `transient--emergency-exit` turns around and itself calls `signal`, calling it in the `debugger` function is a little suspect.

(also, I think you meant `(apply #'debug args)`)

Reply:
> Do you need to call `transient--emergency-exit` before `debug` when you're also adding it to `debugger-mode-hook`?

Yes. Otherwise I would have used `unwind-protect`. The effect of all the transient keymaps and hook functions has to be undone before the user can do things again. It appears that `debug` enters some recursive edit, and whatever comes after the call to `debug` does not evaluated until the user exits that recursive edit, which is not possible to do if the transient keymaps and hooks prevents the user from doing it.

> (also, I think you meant `(apply #'debug args)`)

Yes, though it appears to also work when not doing that.

Reply:
Hm. Wait.

> Do you need to call `transient--emergency-exit` before `debug` when you're also adding it to `debugger-mode-hook`?

I'll have to look at that again but I am not in the mood. (I just tried to help my mum with a computer issue for an hour and using a few dozen photos of hopefully enough of the relevant window for me to make sense of it. And now I am a little depressed. "Click on whatever sounds like 'make it so'. Did something happen? Send me a photo of that." Rinse and repeat for an hour. Aaaaaaaaaaaargh!)